### ISSUE TITLE
Unrelated cleanup functions are called when running an action

### ISSUE THREAD

**Issue by shulhi** [bloodyowl/rescript-react-update#15]
Hi recently I found out about an issue with the reducer cancelling other effects when running some other effect.

For example,

```rescript
let reducer = (state, action) =>
  switch action {
  | Start => ReactUpdate.SideEffects(_ => {
    let intervalId = Js.Global.setInterval(() => {
      Js.log("hello")
    }, 2_000)

    Some(() => Js.Global.clearInterval(intervalId))
  })
  | Run => ReactUpdate.SideEffects(_ => Js.log("Runing"))
  }
}
```

**Expected**
Run `Start` action, the set interval should run the function printing `"hello"` every 2s. 
Then run `Run`, it should prints `"Running"` and the interval should still be running.

**What actually happens**
After running `Run`, the interval stops because the cleanup function for `Start` is called.

So I was expecting that the cleanup function should only be run for that particular action that is being run. It shouldn't run all other cleanup functions.

I was wondering is this is by design or a bug? It would be nice if the cleanup function is only called for the particular action that is being called. 

Some other scenario that would be useful is when an action is running an effect that uses Fetch and in its cleanup function it can abort the fetch using AbortController. If the fetch request is taking a while it will be cancelled when another action is called.

**Comment by bloodyowl** [comment]
Hi!

Nice catch! I've opened #16 to address this.

**Comment by bloodyowl** [comment]
Fix released in 5.0.0

**Comment by shulhi** [comment]
That was fast! Thank you.
